numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -283,6 +283,84 @@ string_isspace_loop(PyArrayMethod_Context *context,
 }
 
 
+template <ENCODING enc>
+static int
+string_islower_loop(PyArrayMethod_Context *context,
+        char *const data[], npy_intp const dimensions[],
+        npy_intp const strides[], NpyAuxData *NPY_UNUSED(auxdata))
+{
+    int elsize = context->descriptors[0]->elsize;
+
+    char *in = data[0];
+    char *out = data[1];
+
+    npy_intp N = dimensions[0];
+
+    while (N--) {
+        Buffer<enc> buf(in, elsize);
+        npy_bool res = buf.islower();
+        *(npy_bool *)out = res;
+
+        in += strides[0];
+        out += strides[1];
+    }
+
+    return 0;
+}
+
+
+template <ENCODING enc>
```

Comment:
Do we have more similar ones?  Cause we could overdo templating to not triplicate the loop part: https://godbolt.org/z/751YrjKP6

Reply:
Good point. Let's merge this and I'll open a new PR that does this for all unary functions.

Reply:
OK with me.  If we go with Marten's path, that is also valid (I am not sure how big the binary or speed differences are, for small strings avoiding the function call may be nice, but overall function calls are usually very low overhead).

Reply:
Actually, turns out that the change was small enough that I felt it was okay for me to push here. Please have a look and let me know if it's okay.

After Marten's PR has been merged I can try it here as well and have a look.